nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -143,26 +165,12 @@ protected void init(final ProcessorInitializationContext context) {
         return properties;
     }
 
-    private int readLines(final InputStream in, final int maxNumLines, final OutputStream out, final boolean keepAllNewLines) throws IOException {
-        int numLines = 0;
-        for (int i = 0; i < maxNumLines; i++) {
-            final long bytes = countBytesToSplitPoint(in, out, keepAllNewLines || (i != maxNumLines - 1));
-            if (bytes <= 0) {
-                return numLines;
-            }
-
-            numLines++;
-        }
-
-        return numLines;
-    }
-
-    private long countBytesToSplitPoint(final InputStream in, final OutputStream out, final boolean includeLineDelimiter) throws IOException {
+    private int readLine(final InputStream in, final OutputStream out,
+                          final boolean includeLineDelimiter) throws IOException {
         int lastByte = -1;
-        long bytesRead = 0L;
+        int bytesRead = 0;
 
         while (true) {
-            in.mark(1);
```

Comment:
Trying to understand the logic here - why was this line removed? It looks like it is marked so that down below, on line 206 we can call in.reset() - with this removed, if we get to that line, where in.reset() is called, I believe it will thrown an IOException.